packages/twenty-front/src/modules/object-record/record-field/hooks/usePersistField.ts

```diff
@@ -31,15 +36,46 @@ import { isFieldRelationValue } from '../types/guards/isFieldRelationValue';
 import { isFieldText } from '../types/guards/isFieldText';
 import { isFieldTextValue } from '../types/guards/isFieldTextValue';
 
-export const usePersistField = () => {
-  const {
-    entityId,
-    fieldDefinition,
-    useUpdateRecord = () => [],
-  } = useContext(FieldContext);
+const useUpsertRecord = () => {
+  const { pendingRecordId } = usePendingRecordId();
```

Comment:
I am not sure this upsert logic should be within PersistField. It does not seem to be the responsability of the field to know about the table row. I would put in in the RecordTableCell